cdap-metadata-ext-spanner/src/test/java/io/cdap/cdap/metadata/spanner/SpannerMetadataStorageTest.java

```diff
@@ -0,0 +1,369 @@
+        } catch (DatabaseNotFoundException e) {
+          // Database not found, proceed.
+        } catch (Exception e) {
+          throw new RuntimeException("Failed to drop database during @AfterClass cleanup", e);
+        }
+      }
+
+      // Delete the instance.
+      if (adminClient != null) {
+        try {
+          adminClient.deleteInstance(INSTANCE_ID);
+        } catch (InstanceNotFoundException e) {
+          // Instance not found, proceed.
+        } catch (Exception e) {
+          throw new RuntimeException("Failed to delete instance during @AfterClass cleanup", e);
+        }
+      }
+
+      // Close the main Spanner client.
+      if (spanner != null) {
+        spanner.close();
+        spanner = null;
+      }
+    }
+  }
+
+  @Before
+  public void beforeTest() throws IOException {
+    spannerMetadataStorage.createIndex();
+  }
+
+  /**
+   * Helper method to simulate initial metadata creation by directly inserting into Spanner.
+   * This is used because `create` and `writeToSpanner` in `SpannerMetadataStorage` are "NOT IMPLEMENTED".
+   * In a real scenario, these would be part of the `MetadataStorage` implementation.
+   */
+  private void simulateInitialMetadata(MetadataEntity entity, Metadata metadata, long version) throws IOException {
+    try {
+      dbClient.readWriteTransaction().run(transaction -> {
+        long currentTime = System.currentTimeMillis();
+        String metadataJson = SpannerMetadataStorage.GSON.toJson(metadata);
+
+        Mutation mutation = Mutation.newInsertOrUpdateBuilder(METADATA_TABLE)
+          .set(Tables.Metadata.METADATA_ID_FIELD).to(spannerMetadataStorage.toMetadataId(entity))
+          .set(Tables.Metadata.NAMESPACE_FIELD).to(entity.getValue(MetadataEntity.NAMESPACE))
+          .set(Tables.Metadata.TYPE_FIELD).to(entity.getType())
+          .set(Tables.Metadata.NAME_FIELD).to(entity.getValue(entity.getType()))
+          .set(Tables.Metadata.CREATED_FIELD).to(currentTime)
+          .set(Tables.Metadata.USER_FIELD).to(metadata.getTags().stream()
+                                                .filter(s -> false).map(ScopedName::getName)
+                                                .collect(Collectors.joining(",")))
+          .set(Tables.Metadata.SYSTEM_FIELD).to(metadata.getTags().stream().
+                                                                      filter(s -> false)
+                                                                      .map(ScopedName::getName)
+                                                  .collect(Collectors.joining(",")))
+          .set(Tables.Metadata.METADATA_COLUMN_FIELD).to(metadataJson)
+          .set(Tables.Metadata.VERSION).to(version)
+          .build();
+        transaction.buffer(mutation);
+        return null; // The callable must return something, null is fine for side effects
+      });
+    } catch (Exception e) {
+      throw new IOException("Failed to setup initial metadata for test: " + e.getMessage(), e);
```

Comment:
nit: No need to use both `e.getMessage` for message and `e` for cause.

Simply: throw new IOException("Failed to setup initial metadata", e);